### PR TITLE
Document internal flake8 plugin schema and configuration

### LINKED ISSUE #55: Is emanspeaks/pyls-flake8 the preferred plugin for flake8 linting?
Hi :wave: 
From reading the [3rd-party plugins section](https://github.com/python-lsp/python-lsp-server#3rd-party-plugins), it seems that the preferred method to lint with `flake8` is by installing the https://github.com/emanspeaks/pyls-flake8 plugin. On the other hand I see there is already a flake8 plugin in this repo: [pylsp/plugins/flake8_lint.py](https://github.com/python-lsp/python-lsp-server/blob/develop/pylsp/plugins/flake8_lint.py). Which one should I use?

If the [pylsp/plugins/flake8_lint.py](https://github.com/python-lsp/python-lsp-server/blob/develop/pylsp/plugins/flake8_lint.py) plugin is the preferred one. I can send a PR to update the [CONFIGURATION.md](https://github.com/python-lsp/python-lsp-server/blob/develop/CONFIGURATION.md) to enable flake8 linting, e.g:

| **Configuration Key** | **Type** | **Description** | **Default** 
|----|----|----|----|
| `pylsp.plugins.flake8.enabled` | `boolean` | Enable or disable the plugin. | `true` |

### PR DESCRIPTION
**Pull request by GerardoGR**
Note-to-self: Take more time to read the comments from the maintainers. I spent 15 mins syncing manually the `schema.json` and the `CONFIGURATION.md` until I realized @krassowski already shared a script to do that automagically :sweat_smile: 

```shell
# from https://github.com/python-lsp/python-lsp-server#development
python scripts/jsonschema2md.py pylsp/config/schema.json CONFIGURATION.md
```

Closes #55

### CODE REVIEW

File: CONFIGURATION.md [hunk 1]
```diff
@@ -1,9 +1,19 @@
 # Python Language Server Configuration
 This server can be configured using `workspace/didChangeConfiguration` method. Each configuration option is described below:
 
-| **Configuration Key** | **Type** | **Description** | **Default** 
+| **Configuration Key** | **Type** | **Description** | **Default**
```
Comment (krassowski) [inline CONFIGURATION.md]: Was this space added in manually? The CI will complain if this file is not exactly as generated by the script.

File: pylsp/config/schema.json [hunk 2]
```diff
@@ -14,6 +14,56 @@
...
+      "default": null,
+      "description": "Hang closing bracket instead of matching indentation of opening bracket's line."
+    },
+    "pylsp.plugins.flake8.ignore": {
+      "type": "array",
+      "default": null,
+      "description": "List of errors and warnings to ignore (or skip)."
+    },
+    "pylsp.plugins.flake8.maxLineLength": {
+      "type": "int",
```
Comment (krassowski) [inline pylsp/config/schema.json]:
```suggestion
      "type": "integer",
```

File: CONFIGURATION.md (hunk 1 above)
Comment (GerardoGR) [reply CONFIGURATION.md]: This was probably my editor removing leading spaces. I've regenerated the file with the script.

**Comment by andfoy** [comment]
@krassowski, is this one ready to merge?

File: pylsp/config/schema.json [hunk 3]
```diff
@@ -14,6 +14,56 @@
...
+    },
+    "pylsp.plugins.flake8.maxLineLength": {
+      "type": "integer",
+      "default": null,
+      "description": "Maximum allowed line length for the entirety of this run."
+    },
+    "pylsp.plugins.flake8.perFileIgnores": {
+      "type": "array",
+      "default": null,
+      "description": "A pairing of filenames and violation codes that defines which violations to ignore in a particular file."
```
Comment (krassowski) [inline pylsp/config/schema.json]: It's not clear to me what the pairing means, I had to got to the ode to understand, so I left a suggestion to add an example (please free to expand or give a different one).
```suggestion
      "description": "A pairing of filenames and violation codes that defines which violations to ignore in a particular file, for example: `[\"file_path.py:W305,W304\"]`)."
```

**Comment by krassowski** [comment]
Yes, it looks good to me, I tested most of these config optoins locally to make sure they work. I added one suggestion for improvement (note: committing in GitHub UI is not enough, this will need regenerating the md file, I can do that) but this can be also done in follow up PR if you were aiming to get this in and cut a release now.

**Review by krassowski — APPROVED** [review]

**Comment by andfoy** [comment]
No worries, we need to finish https://github.com/python-lsp/python-lsp-server/pull/53 before making the release

**Pull request merged**